ddtrace/contrib/openai/patch.py

```diff
@@ -0,0 +1,464 @@
+import os
+import random
+import sys
+
+from ddtrace import config
+from ddtrace.internal.constants import COMPONENT
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.utils.formats import asbool
+
+from . import _log as ddlogs
+from .. import trace_utils
+from .. import trace_utils_async
+from ...pin import Pin
+from ..trace_utils import wrap
```

Comment:
I think that going forward we want to use `ddtrace.internal.wrapping.wrap` for pure-Python functions (openai-python seems to be pure-Python)

Reply:
@P403n1x87 I attempted to do it but `ddtrace.internal.wrapping.wrap` isn't compatible with our other tracing utils like `with_traced_module`. We should look to see if we can replace the underlying implementation of `trace_utils.wrap()` with `ddtrace.internal.wrapping.wrap`.